delivery_carrier_seur/model/stock.py

```diff
@@ -0,0 +1,167 @@
+
+_logger = logging.getLogger(__name__)
+try:
+    from seur.picking import Picking
+except ImportError:
+    _logger.debug('Can not `from seur.picking import Picking`.')
+
+
+class ShippingLabel(models.Model):
+    _inherit = 'shipping.label'
+
+    @api.model
+    def _get_file_type_selection(self):
+        """ To inherit to add file type """
+        res = super(ShippingLabel, self)._get_file_type_selection()
+        res.append(('txt', 'TXT'))
+        return res
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    seur_service_code = fields.Selection(
+        selection='_get_seur_services', string='Seur Service Code',
+        default=False)
+    seur_product_code = fields.Selection(
+        selection='_get_seur_products', string='Seur Product Code',
+        default=False)
+
+    def _get_seur_services(self):
+        return self.env['delivery.carrier'].SEUR_SERVICES
+
+    def _get_seur_products(self):
+        return self.env['delivery.carrier'].SEUR_PRODUCTS
+
+    @api.onchange('carrier_id')
+    def carrier_id_onchange(self):
+        if not self.carrier_id:
+            return
+        carrier = self.carrier_id
+        self.seur_service_code = carrier.seur_service_code
+        self.seur_product_code = carrier.seur_product_code
+
+    @api.multi
+    def _generate_seur_label(self, package_ids=None):
+        self.ensure_one()
+        if not self.carrier_id.seur_config_id:
+            raise exceptions.Warning(_('No SEUR Config defined in carrier'))
+        if not self.picking_type_id.warehouse_id.partner_id:
+            raise exceptions.Warning(
+                _('Please define an address in the %s warehouse') % (
+                    self.warehouse_id.name))
+
+        config = self.carrier_id.seur_config_id
+
+        seur_context = {
+            'printer': 'ZEBRA',
+            'printer_model': 'LP2844-Z',
+            'ecb_code': '2C',
+        }
+
+        if config.file_type == 'pdf':
+            seur_context['pdf'] = True
+
+        seur_picking = Picking(
+            config.username,
+            config.password,
+            config.vat,
+            config.franchise_code,
+            'Odoo',  # seurid
+            config.integration_code,
+            config.accounting_code,
+            seur_context
+        )
+
+        try:
+            connect = seur_picking.test_connection()
+            if connect != 'Connection successfully':
+                raise exceptions.Warning(
+                    _('Error conecting with SEUR:\n%s' % connect))
+        except HTTPError, e:
+            raise exceptions.Warning(
+                _('Error conecting with SEUR try later:\n%s' % e))
+
+        data = self._get_label_data()
+        tracking_ref, label, error = seur_picking.create(data)
+
+        if error:
+            raise exceptions.Warning(
+                _('Error sending label to SEUR\n%s' % error))
+
+        self.write({'carrier_tracking_ref': tracking_ref})
```

Comment:
as you have only one record, here you can write 
`self.carrier_tracking_ref = tracking_ref`